sds_data_manager/lambda_code/batch_starter.py

```diff
@@ -1,433 +1,364 @@
-        instrument = item["instrument"]
-        level = item["level"]
-        date = item["date"]
-        grouped_data[instrument][level].append(date)
+    components = extract_components(filename)
+
+    instrument = components["instrument"]
+    level = components["datalevel"]
+    start_date = components["startdate"]
+
+    format_start_date = datetime.strptime(start_date, "%Y%m%d")
+
+    # Format year and month from the datetime object
+    year = format_start_date.strftime("%Y")
+    month = format_start_date.strftime("%m")
 
-    grouped_dict = {
-        instrument: dict(levels) for instrument, levels in grouped_data.items()
-    }
+    # Base S3 path
+    s3_base_path = f"s3://{s3_bucket}/imap/{instrument}/{level}/{year}/{month}/"
 
-    return grouped_dict
+    # Prepare the final command
+    # Pre-construct parts of the string
+    instrument_part = f"--instrument {instrument}"
+    level_part = f"--level {level}"
+    s3_uri = f"--s3_uri '{s3_base_path}{filename}'"
+    dependency_part = f"--dependency {upstream_dependencies}"
 
+    prepared_data = f"{instrument_part} {level_part} {s3_uri} {dependency_part}"
 
-def get_downstream_dependencies(key):
+    return prepared_data
+
+
+def extract_components(filename: str):
     """
-    Retrieves downstream dependencies of a given instrument.
+    Extracts components from filename.
 
     Parameters
     ----------
-    key : str
-        The key from the JSON file.
+    filename : str
+        Path of dependency data.
 
     Returns
     -------
-    dict
-        The value associated with the provided key in the JSON file.
-    """
-
-    # Construct the path to the JSON file
-    dir_path = os.path.dirname(os.path.realpath(__file__))
-    dependency_path = os.path.join(dir_path, "downstream_dependents.json")
-
-    with open(dependency_path) as file:
-        data = json.load(file)
-        value = data.get(key)
-
-        if value is None:
-            raise KeyError(f"Key '{key}' not found in the JSON file.")
+    components : dict
+        Dictionary containing components.
 
-        return value
+    """
+    pattern = (
+        r"^imap_"
+        r"(?P<instrument>[^_]*)_"
+        r"(?P<datalevel>[^_]*)_"
+        r"(?P<descriptor>[^_]*)_"
+        r"(?P<startdate>\d{8})_"
+        r"(?P<enddate>\d{8})_"
+        r"(?P<version>v\d{2}-\d{2})"
+        r"\.cdf$"
+    )
+    match = re.match(pattern, filename)
+    components = match.groupdict()
+    return components
 
 
 def lambda_handler(event: dict, context):
     """Handler function"""
     logger.info(f"Event: {event}")
     logger.info(f"Context: {context}")
 
-    # Event details (TBD)
-    instrument = event["detail"]["object"]["instrument"]
-    instrument_downstream = get_downstream_dependencies(instrument)
-    db_secret_arn = os.environ.get("SECRET_ARN")
+    # Event details:
+    filename = event["detail"]["object"]["key"]
+    components = extract_components(filename)
+    instrument = components["instrument"]
```

Comment:
Don't worry about it for this PR. In upcoming PR, we can replace this with path_helper.py class. That's in review too.